### PR TITLE
Blog: Setting up Node-RED on Raspberry Pi

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

File: src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+- **Affordable:** Raspberry Pi boards are cost-effective, making them accessible for individuals, educators, and hobbyists.
+- **Versatility:** Raspberry Pi supports various operating systems and can be used for a wide range of projects, from basic computing tasks to advanced IoT applications.
+- **GPIO Pins:** The GPIO (General Purpose Input/Output) pins allow for hardware connections, enabling users to interface with sensors, motors, and other devices.
+- **Compact Size:** Raspberry Pi boards are small in size, making them convenient for embedding into projects with limited space.
+
+## Installing Flowfuse Device Agent on Raspberry Pi
+
+The FlowFuse Device Agent, developed by FlowFuse, simplifies the operation of Node-RED on your edge device, facilitating connections and interactions with hardware devices. Furthermore, it enables remote management, streamlining the monitoring and administration of the Node-RED application running on the device from any remote location.
```
Comment (Hasmin-AC) [inline src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md:28]:
```suggestion
[The FlowFuse Device Agent](https://flowfuse.com/docs/device-agent/introduction/), developed by FlowFuse, simplifies the operation of Node-RED on your edge device, facilitating connections and interactions with hardware devices. Furthermore, it enables remote management, streamlining the monitoring and administration of the Node-RED application running on the device from any remote location.
```
```suggestion
The FlowFuse Device Agent, developed by FlowFuse, simplifies the operation of Node-RED on your edge device, facilitating connections and interactions with hardware devices. Furthermore, it enables remote management, streamlining the monitoring and administration of the Node-RED application running on the device from any remote location.
```

File: src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+{% renderFlow %}
+[{"id":"bdb27b9ab0d94897","type":"inject","z":"b152a914653d9fce","name":"Inject","props":[{"p":"payload"}],"repeat":"5","crontab":"","once":true,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":330,"y":380,"wires":[["4ece54ca3c91f7ff"]]},{"id":"4ece54ca3c91f7ff","type":"debug","z":"b152a914653d9fce","name":"debug 1","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":580,"y":380,"wires":[]}]
+{% endrenderFlow %}
+
+Now, your basic flow is set up. When you click the "Inject" button, it will trigger a message to be sent to the "Debug" node, which will display the message in the debug panel. This helps you understand how nodes can be connected to create a flow in Node-RED. 
+
+## Securing Node-RED
+
+When using the Node-RED along with FlowFuse , there's no need to add security measures as FlowFuse provides robust security, ensuring that no one can access your device editor without proper authorization. However, disabling the editor after development prevents unauthorized changes to your flows by team members.
```
Comment (Hasmin-AC) [inline src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md:101]:
```suggestion
When using Node-RED along with FlowFuse , there's no need to add security measures as FlowFuse provides robust security, ensuring that no one can access your device editor without proper authorization. However, disabling the editor after development prevents unauthorized changes to your flows by team members.
```
```suggestion
When using the Node-RED along with FlowFuse , there's no need to add security measures as FlowFuse provides robust security, ensuring that no one can access your device editor without proper authorization. However, disabling the editor after development prevents unauthorized changes to your flows by team members.
```

**Review by Hasmin-AC — APPROVED** [review]

File: src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+### Reading sensor data with Node-RED
+
+#### Installing bcm2835 library 
+
+The bcm2835 library is essential for Raspberry Pi (RPi) as it provides access to GPIO and other IO functions on the Broadcom BCM 2835 chip, used in the Raspberry Pi 4. This allows control and interface with various external devices connected to the GPIO pins on the RPi 4 boards.
+
+1. Open the command line of your Raspberry Pi.
+2. Navigate to your home directory by entering the following command:
+
+`
```
Comment (ZJvandeWeg) [inline src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md:141]: If it's a block, go with triple backticks

File: src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+- **Affordable:** Raspberry Pi boards are cost-effective, making them accessible for individuals, educators, and hobbyists.
+- **Versatility:** Raspberry Pi supports various operating systems and can be used for a wide range of projects, from basic computing tasks to advanced IoT applications.
+- **GPIO Pins:** The GPIO (General Purpose Input/Output) pins allow for hardware connections, enabling users to interface with sensors, motors, and other devices.
+- **Compact Size:** Raspberry Pi boards are small in size, making them convenient for embedding into projects with limited space.
+
+## Installing Flowfuse Device Agent on Raspberry Pi
+
+[The FlowFuse Device Agent](https://flowfuse.com/docs/device-agent/introduction/), developed by FlowFuse, simplifies the operation of Node-RED on your edge device, facilitating connections and interactions with hardware devices. Furthermore, it enables remote management, streamlining the monitoring and administration of the Node-RED application running on the device from any remote location.
```
Comment (ZJvandeWeg) [inline src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md:28]:
```suggestion
[The FlowFuse Device Agent](/docs/device-agent/introduction/), developed by FlowFuse, simplifies the operation of Node-RED on your edge device, facilitating connections and interactions with hardware devices. Furthermore, it enables remote management, streamlining the monitoring and administration of the Node-RED application running on the device from any remote location.
```

File: src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+### Prerequisites for Installation:
+
+- Required Hardware:
+
+Before proceeding further, ensure you have all the necessary hardware components ready. This includes the Raspberry Pi single-board computer (recommended models: Raspberry Pi version 1 or higher ), an SD card (8GB or larger recommended), a compatible power supply, and any required peripherals such as a keyboard, mouse, and display. It's also beneficial to have a DHT11 sensor for following the practical guide further, but it is not necessary for the installation process.
```
Comment (ZJvandeWeg) [inline src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md:34]: Many RPIs run in headless mode, through SSH. No need for keyboard, mouse, or display

```suggestion
Before proceeding further, ensure you have all the necessary hardware components ready. This includes the Raspberry Pi single-board computer (recommended models: Raspberry Pi version 1 or higher ), an SD card (8GB or larger recommended), a compatible power supply, and any required peripherals such as a keyboard, mouse, and display. It's also beneficial to have a DHT11 sensor for following the practical guide further, but it is not necessary for the installation process.
```

File: src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+
+### Prerequisites for Installation:
+
+- Required Hardware:
+
+Before proceeding further, ensure you have all the necessary hardware components ready. This includes the Raspberry Pi single-board computer (recommended models: Raspberry Pi version 1 or higher ), an SD card (8GB or larger recommended), a compatible power supply, and any required peripherals such as a keyboard, mouse, and display. It's also beneficial to have a DHT11 sensor for following the practical guide further, but it is not necessary for the installation process. 
+
+- Software Preparation:
+
+Ensure your Raspberry Pi is fully set up and running on the latest version of Raspbian OS. For more information on installing Raspberry Pi OS, refer to [How to Install Raspberry Pi OS on Your Pi](https://raspberrytips.com/install-raspberry-pi-os/). Additionally, ensure your Raspberry Pi is connected to a stable network, either via Ethernet or Wi-Fi.
```
Comment (ZJvandeWeg) [inline src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md:38]: A debian based OS will do? Raspbian is recommended, I agree, but not required.

File: src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md
Comment (ZJvandeWeg) [inline src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md:1]: Maybe at: `/node-red/hardware/raspberry-pi`? cc @Yndira-E

File: src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+
+If you are unsure about the pin configurations, you can refer to the images below:
+
+!["Image showing the Raspberry Pi pins with diagram"](./images/how-to-setup-node-red-on-raspberry-pi-pins.png "Image showing the Raspberry Pi pins with diagram"){data-zoomable}
+
+### Reading sensor data with Node-RED
+
+#### Installing bcm2835 library 
+
+The bcm2835 library is essential for Raspberry Pi (RPi) as it provides access to GPIO and other IO functions on the Broadcom BCM 2835 chip, used in the Raspberry Pi 4. This allows control and interface with various external devices connected to the GPIO pins on the RPi 4 boards.
```
Comment (ZJvandeWeg) [inline src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md:139]: What happens on the RPI5?

File: src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+2. Navigate to your home directory by entering the following command:
+
+```
+cd ~
+```
+
+3. Download the bcm2835 library package by executing the following command:
+
+```
+wget  http://www.airspayce.com/mikem/bcm2835/bcm2835-1.75.tar.gz
```
Comment (ZJvandeWeg) [inline src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md:151]: `apt-get install` doens't have this package? It feels weird to advise to wget a file. 

Does `wget` get installed by default? Or is using `curl` better?

File: src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md (hunk 7 above)
Comment (sumitshinde-84) [reply src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md:139]: This dht sensor node requires the bcm2835 library to be preinstalled, which only works on Raspberry Pi models up to version 4

File: src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md (hunk 8 above)
Comment (sumitshinde-84) [reply src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md:151]: No, this library only gets installed with `wget`.

File: src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md
Comment (Yndira-E) [reply src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md:1]: This will save us a future redirect, so I second your motion 😉

Comment (ZJvandeWeg) [reply src/blog/2024/05/how-to-setup-node-red-on-raspberry-pi.md:1]: @sumitshinde-84 Can you make this happen?

File: src/node-red/hardware/index.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+  key: Hardware
+meta:
+   title: Setting Up Node-RED on Different Hardware
+   description: Learn how to set up Node-RED on various hardware platforms, including Raspberry Pi, Arduino, and more.
+   keywords: node-red, node-red with different hardwares, flowfuse device agent
+---
+
+# Hardware
+
+Welcome to the Hardware category. Here, we provide comprehensive resources and guides for installing and configuring Node-RED on various hardware platforms. Whether you're using a Raspberry Pi, windows , ctrlX, or other devices, you'll find the information you need to get started.
```
Comment (ZJvandeWeg) [inline src/node-red/hardware/index.md:12]:
```suggestion
FlowFuse supports a wide range of hardware devices to allow users to connect Node-RED to many devices and sensors. This allows for the creation of interactive and automated systems, where data from physical inputs can trigger actions or responses in the digital realm. By bridging the gap between software and hardware, Node-RED expands its applicability to a wide range of use cases.
```

**Pull request merged**